File: JInstantSearch/app/src/main/java/com/woowrale/architecture/mvp/ui/search/local/LocalSearchView.java
package com.woowrale.architecture.mvp.ui.search.local;

import com.woowrale.architecture.mvp.data.model.Contact;
import com.woowrale.architecture.mvp.ui.adapters.ContactsAdapterFilterable;
import com.woowrale.architecture.mvp.ui.base.BasePresenter;

import java.util.List;

import io.reactivex.disposables.CompositeDisposable;

public interface LocalSearchView extends BasePresenter.View {

    CompositeDisposable getDisposable();

    ContactsAdapterFilterable getmAdapter();

    List<Contact> getContactsList();

}

File: AndroidPatternsApp/kmvp-app/src/main/java/com/woowrale/mvp/ui/search/local/LocalSearchView.kt
package com.woowrale.mvp.ui.search.local

import com.woowrale.mvp.data.model.Contact
import com.woowrale.mvp.ui.adapters.ContactsAdapterFilterable
import com.woowrale.mvp.ui.base.BasePresenter
import io.reactivex.disposables.CompositeDisposable

interface LocalSearchView: BasePresenter.View{

    fun getDisposable(): CompositeDisposable

    fun getmAdapter(): ContactsAdapterFilterable

    fun getContactsList(): List<Contact>

    fun setContactsList(contacts: List<Contact>)
}